packages/runtime/lifecycle/src/api.ts

```diff
@@ -32,33 +45,76 @@ export type ApplyInputsRequest = {
   invalidHereafter?: ISO8601;
 };
 /**
- * TODO: comment
+ * This Interface provides capabilities for runnning a Contract over Cardano.
  */
 export interface ContractsAPI {
   /**
-   * TODO: comment
+   * Submit to the Cardano Ledger, the Transaction(Tx) that will create the Marlowe Contract passed in the request.
+   * @param request Request parameters for creating a Marlowe Contract on Cardano
    * @throws DecodingError
    */
-  create(req: CreateRequest): Promise<ContractId>;
-  // [[apply-inputs-next-provider]]
-  // DISCUSSION: Instead of providing a function from "Next" => ApplyInputsRequest, I think
-  //             this as a low level API should just receive the ApplyInputsRequest, and if we want we can
-  //             add a "higher-level" helper
-  //             Search for [apply-inputs-next-provider] for usage
+  submitCreateTx(request: CreateRequest): Promise<[ContractId, TxId]>;
+
   /**
-   * TODO: comment
+   * @experimental
+   * Submit to the Cardano Ledger & Wait Confirmation, the Transaction(Tx) that will create the Marlowe Contract passed in the request.
+   * It's higher level capabilities compared to `submitCreateTx`.
+   * It combines `submitCreateTx` & waitConfirmation for you to abstract Cardano infrastructure concerns.
+   * @param request Request parameters for creating a Marlowe Contract on Cardano
+   * @throws DecodingError
+   */
+  create(request: CreateRequest): Promise<ContractId>;
+
+  /**
+   * Submit to the Cardano Ledger, the Transaction(Tx) that will apply inputs to a given created contract (see `submitCreateTx`).
+   * It's lower level version of `applyInputs`.
+   * @param contractId Contract Id where inputs will be applied
+   * @param inputsRequest inputs to apply
+   * @throws DecodingError
+   */
+  submitApplyInputsTx(
+    contractId: ContractId,
+    inputsRequest: ApplyInputsRequest
+  ): Promise<TxId>;
+
+  /**
+   * Provide Applicable Inputs and Reducibility Information for a given contract for the connected wallet.
+   * @param contractId Contract Id of a created contract
+   * @throws DecodingError
+   */
+  getNext(contractId: ContractId): Promise<Next>;
```

Comment:
Maybe rename to `getNextActions`?

Reply:
It is different from the concept of actions, it gives you all the applicable inputs variations...

Reply:
Like not really actions, and not really inputs... something in the middle...